kolena/annotation.py

```diff
@@ -382,6 +383,67 @@ class ScoredLabeledTimeSegment(TimeSegment):
     """The score associated with this time segment."""
 
 
+@dataclass(frozen=True, config=ValidatorConfig)
+class TextSegment(Annotation):
+    """
+    Segment of text in the associated text field
+
+    ```py
+    text_segments: List[TextSegment] = [
+        TextSegment(text_field="text", start=0, end=5),
+        TextSegment(text_field="summary", start=10, end=51),
+    ]
+    """
+
+    text_field: str
+    """Text field column name containing the text segments."""
+
+    start: int
+    """Start index of the text segment."""
+
+    end: int
+    """End index of the text segment."""
+
+    def __post_init__(self) -> None:
+        if self.start < 0:
+            raise ValueError(f"Start index must be non-negative ({self.start} provided)")
+        if self.end < 0:
+            raise ValueError(f"End index must be non-negative ({self.end} provided)")
+        if self.start > self.end:
+            raise ValueError(f"Start index must be less than or equal to end index ({self.start} > {self.end})")
```

Comment:
what does it mean to have `start == end`?

Reply:
Zero length segment, probably not useful as well